Datagrid/ProxyQuery.php

```diff
@@ -328,6 +328,18 @@ protected function getFixedQueryBuilder(QueryBuilder $queryBuilder)
             $queryBuilderId->addSelect($sortBy);
         }
 
+        // For any orderBy clause defined directly in the dqlParts
+        $dqlParts = $queryBuilderId->getDqlParts();
+        if ($dqlParts['orderBy'] && count($dqlParts['orderBy'])) {
+            $sqlOrderColumns = array();
+            foreach ($dqlParts['orderBy'] as $part) {
+                foreach ($part->getParts() as $orderBy) {
+                    $sqlOrderColumns[] = preg_replace("/\s+(ASC|DESC)$/i", '', $orderBy);
+                }
+            }
+            $queryBuilderId->addSelect(implode(', ', $sqlOrderColumns));
```

Comment:
A comment to end your sentence line 331 would be nice. Something like `// get rid of the direction and add what should be the column to the select clause`